### PR TITLE
feat: add cucumber tests for leader failure

### PR DESCRIPTION
**Pull request by jorgeantonio21**
Description
---
Adds a cucumber test for leader failure, with a 4-vn committee.

Motivation and Context
---

How Has This Been Tested?
---
Integration tests

### CODE REVIEW

File: dan_layer/storage_sqlite/src/sqlite_shard_store_factory.rs [hunk 1]
```diff
@@ -730,7 +730,9 @@ impl ShardStoreReadTransaction<PublicKey, TariDanPayload> for SqliteShardStoreRe
         .map_err(|e| StorageError::QueryError {
             reason: format!("Get transaction: {}", e),
         })?;
-        Ok(res.into_iter().map(|transaction| transaction.into()).collect())
+        let res = res.into_iter().map(|transaction| transaction.into()).collect();
+        println!("FLAG CUCUMBER: THE DATABASE RES = {:?}", res);
```
Comment (sdbondi) [inline dan_layer/storage_sqlite/src/sqlite_shard_store_factory.rs]:
```suggestion

```

File: applications/tari_validator_node_web_ui/src/routes/Transaction/Components/Output.tsx.orig [hunk 2]
```diff
@@ -55,59 +55,6 @@ export default function Output({
           <TableBody>
             {output.map((row) => {
               let justify = JSON.parse(row.justify);
```
Comment (sdbondi) [inline applications/tari_validator_node_web_ui/src/routes/Transaction/Components/Output.tsx.orig]: These .orig files are from conflicts and can be removed

File: applications/tari_validator_node_web_ui/src/routes/VN/Components/RecentTransactions.tsx.orig [hunk 3]
```diff
@@ -158,22 +158,6 @@ function RecentTransactions() {
   useEffect(() => {
     getRecentTransactions().then((recentTransactions) => {
       setRecentTransacations(
-<<<<<<< HEAD
```
Comment (sdbondi) [inline applications/tari_validator_node_web_ui/src/routes/VN/Components/RecentTransactions.tsx.orig:161]: Can remove this file

File: applications/tari_validator_node/tests/cucumber.rs [hunk 4]
```diff
@@ -396,6 +546,33 @@ async fn wait_seconds(_world: &mut TariWorld, seconds: u64) {
     tokio::time::sleep(Duration::from_secs(seconds)).await;
 }
 
+#[then(expr = "the transaction succeeds")]
```
Comment (sdbondi) [inline applications/tari_validator_node/tests/cucumber.rs]: If there are more than committee_size (7) VNs this won't work currently but this is fine for now
```suggestion
#[then(expr = "all transactions succeed on all validator nodes")]
```

**Review by sdbondi — COMMENTED** [review]
Nice, just need to get rid of the println! in the database layer

File: dan_layer/storage_sqlite/src/sqlite_shard_store_factory.rs (hunk 1 above)
Comment (jorgeantonio21) [reply dan_layer/storage_sqlite/src/sqlite_shard_store_factory.rs]: Nice catch :) thanks !

File: applications/tari_validator_node_web_ui/src/routes/VN/Components/RecentTransactions.tsx.orig (hunk 3 above)
Comment (jorgeantonio21) [reply applications/tari_validator_node_web_ui/src/routes/VN/Components/RecentTransactions.tsx.orig:161]: Removed :)

**Review by stringhandler — APPROVED** [review]

**Pull request merged**